addon/-private/column-tree.js

```diff
@@ -0,0 +1,783 @@
+/* global Ember */
+import EmberObject, { get, set } from '@ember/object';
+import ObjectProxy from '@ember/object/proxy';
+import { addObserver, removeObserver } from '@ember/object/observers';
+import { A as emberA } from '@ember/array';
+
+import { computed } from '@ember-decorators/object';
+import { readOnly } from '@ember-decorators/object/computed';
+
+import { scheduler, Token } from 'ember-raf-scheduler';
+
+import { getOrCreate } from './meta-cache';
+import { move, splice, mergeSort } from './utils/array';
+import { getScale, getOuterClientRect, getInnerClientRect } from './utils/element';
+import { MainIndicator, DropIndicator } from './utils/reorder-indicators';
+
+const SCROLL_THRESHOLD = 50;
+
+const DEFAULT_COLUMN_WIDTH = 100;
+const DEFAULT_MIN_WIDTH = 50;
+const DEFAULT_MAX_WIDTH = Infinity;
+
+export const RESIZE_MODE = {
+  STANDARD: 'standard',
+  FLUID: 'fluid',
+};
+
+export const FILL_MODE = {
+  EQUAL_COLUMN: 'equal-column',
+  FIRST_COLUMN: 'first-column',
+  LAST_COLUMN: 'last-column',
+};
+
+export const WIDTH_CONSTRAINT = {
+  NONE: 'none',
+  EQ_CONTAINER: 'eq-container',
+  GTE_CONTAINER: 'gte-container',
+  LTE_CONTAINER: 'lte-container',
+};
+
+class TableColumnMeta extends EmberObject {
+  @readOnly('_node.isLeaf') isLeaf;
+  @readOnly('_node.isFixed') isFixed;
+  @readOnly('_node.width') width;
+  @readOnly('_node.offsetLeft') offsetLeft;
+  @readOnly('_node.offsetRight') offsetRight;
+}
+
+class TableColumnProxy extends ObjectProxy {
+  @computed('content')
+  get meta() {
+    return getOrCreate(get(this, 'content'), get(this, '_cache'), TableColumnMeta);
+  }
+}
+
+/**
+  Single node of a ColumnTree
+*/
+class ColumnTreeNode {
+  constructor(column, tree, parent) {
+    this.tree = tree;
+
+    if (!parent) {
+      this.column = column;
+      this.isRoot = true;
+    } else {
+      this.parent = parent;
+      this.column = TableColumnProxy.create({ content: column, _cache: get(tree, 'metaCache') });
+
+      let meta = get(this, 'column.meta');
+
+      set(meta, '_node', this);
+      meta.registerElement = this.registerElement.bind(this);
+
+      meta.startResize = tree.startResize.bind(tree, this);
+      meta.updateResize = tree.updateResize.bind(tree, this);
+      meta.endResize = tree.endResize.bind(tree, this);
+
+      meta.startReorder = tree.startReorder.bind(tree, this);
+      meta.updateReorder = tree.updateReorder.bind(tree, this);
+      meta.endReorder = tree.endReorder.bind(tree, this);
```

Comment:
Equivalent to `meta.endReorder = () => tree.endReorder(this)`. I prefer the closure but I don't have any convincing justification for changing the version already written here.